### DISCUSSION TITLE
About heap collection

### DISCUSSION THREAD

**Discussion by tellmewhattodo** [glowroot/glowroot#1230] [Q&A] [answered]
does Glowroot agent collect heap information automatically?
while I know Glowroot do it when I request it via dashboard, but I'd like to know if Glowroot will do it in background.
it concerned me, because I noticed some strange JVM pause with glowroot on, which is not paused by GC.

**Comment by JayceLau** [comment]
+1

**Comment by Never-lab** [comment, marked as answer]
## Triage

**Q&A** — not a bug.

Glowroot does **not** collect heap dumps or heap histograms in the background. Both are **on-demand only** via JVM tab (`/backend/jvm/heap-dump`, `/backend/jvm/heap-histogram` in `JvmJsonService` / `LiveJvmServiceImpl`).

Background overhead comes from trace capture, profiling (if enabled), gauge polling, and H2 writes — not automatic heap collection.

JVM pauses without GC logs may be safepoints from profiling, H2 compaction, or weaving — worth checking `profilingIntervalMillis` and embedded DB size.

**Verdict:** convert to **Discussions → Q&A** (wrong channel for this issue).

---
*Community triage*
